pandas-stubs/_typing.pyi

```diff
@@ -253,9 +253,11 @@ XMLParsers: TypeAlias = Literal["lxml", "etree"]
 
 # Any plain Python or numpy function
 Function: TypeAlias = Union[np.ufunc, Callable[..., Any]]
+# Use custom HashableT in shared types to avoid conflicts with shared HashableTs
+_GroupByObjectHashableT = TypeVar("_GroupByObjectHashableT", bound=Hashable)
```

Comment:
How about moving it to `_typing` and calling it `HashableTa` or something like that? I'm sure we need it also in other places.

Reply:
The purpose of this one is that it will never be used outside for _typing.  This makes sure that it doesn't accidentally get used in a definition with one of the other HashableT instances.  I've renamed it to be simpler.

Reply:
I see your point, and have now used a public name.

Reply:
I like your idea of giving it a private name, Could have something like `_HashableT` or `_HashableTa` in `_typing`